### DISCUSSION TITLE
Dependency resolver

### DISCUSSION THREAD

**Discussion by AdelKS** [pkgcraft/pkgcraft#58] [General]
Hello,

I've been working for a while on [a dependency resolver](https://github.com/AdelKS/QuantumResolver/) for Gentoo, as a hobby project, I want to know if I can do something as feature complete and hopefully replace emerge's back-end with something faster (everything started from the fact that `emerge`'s dependency resolution is slow. And since it didn't get replaced, there still isn't any better alternative). 

I have nearly programmed all the building blocks (retrieve all the data and represent it internally) to be able to start working on the dependency graph. I have a few ideas, and I will soon start implementing that in practice, I will most certainly make visualization tools along the way to better understand how things are connected with each other etc.

I do not have much to offer yet, I just wanted to :wave: (antarus redirected me to you guys), hopefully we can interact, work with each other, maybe interface our projects. I do not know if you guys have already written a dependency resolver and/or are interested in that.

**Comment by radhermit** [comment]
Sounds interesting, but I'm not sure how much the projects will be able to share other than any algorithm ideas/designs since I assume you're going to keep your implementation in C++ and target dropping into portage as a high level resolver replacement rather then rewriting more of the underlying work.

I'd prefer keeping pkgcraft core code in rust (not wrapping any external libraries) and it targets replacing the entire stack including some wild and crazy ideas, e.g. it uses its own fork of bash in order to leverage bash builtins for all internal commands, allow bash process reuse, and for several other reasons.

However, if you're familiar with IRC and ever want to chat about related topics, pkgcraft now has an irc channel at #pkgcraft on libera.
